src/appleseed/renderer/modeling/camera/camera.cpp

```diff
@@ -311,44 +368,218 @@ double Camera::extract_near_z() const
+                                                        );
+
+    m_shutter_curve_nonlinear_open_cdf[5] =     0.6f *  (
+                                                        2.0f * (t01 + ot - 2.0f*t00) -
+                                                        6.0f*s00*(oet - 2.0f*ot + 5.0f*t00 - 4.0f*t01) +
+                                                        3.0f*s01*(oet - 3.0f*ot + 7.0f*t00 - 5.0f*t01)
+                                                        );
+
+    m_shutter_curve_nonlinear_open_cdf[6] =     0.5f * (3.0f*s00 - 3.0f*s01 + 1.0f) *
+                                                (oet - ot + 3.0f*t00 - 3.0f*t01);
+
+    m_shutter_curve_nonlinear_close_cdf[0] =    m_shutter_curve_nonlinear_open_cdf.evaluate(1.0f) + cst - oet;
+
+    m_shutter_curve_nonlinear_close_cdf[1] =    3.0f * (t10-cst);
+
+    m_shutter_curve_nonlinear_close_cdf[2] =    1.5f *  (
+                                                        cst * (5.0f - 3.0f*s10) +
+                                                        t10 * (3.0f*s10 - 7.0f) +
+                                                        2.0f*t11
+                                                        );
+
+    m_shutter_curve_nonlinear_close_cdf[3] =    cst * (12.0f*s10 - 3.0f*s11 - 10.0f) +
+                                                t10 * (3.0f*s11 -18.0f*s10 + 18) +
+                                                t11 * (6.0f*s10 - 9.0f) +
+                                                ct;
+
+    m_shutter_curve_nonlinear_close_cdf[4] =    0.75f * (
+                                                        cst * (9.0f*s11 -18.0f*s10 + 10.0f) +
+                                                        ct * (3.0f*s10 - 3.0f) +
+                                                        t10 * (36.0f*s10 - 15.0f*s11 - 22.0f) +
+                                                        t11 * (6.0f*s11 -21.0f*s10 + 15.0f)
+                                                        );
+
+    m_shutter_curve_nonlinear_close_cdf[5] =    0.6f *  (
+                                                        cst * (12*s10 - 9*s11 - 5.0f) +
+                                                        ct * (3.0f*s11 - 6.0f*s10 + 3.0f) +
+                                                        t10 * (21.0f*s11 - 30.0f*s10 + 13.0f) +
+                                                        t11 * (24.0f*s10 - 15.0f*s11 - 11.0f)
+                                                        );
+
+    m_shutter_curve_nonlinear_close_cdf[6] =    0.5f *
+                                                (3.0f*s10 - 3.0f*s11 - 1.0f) *
+                                                (ct - cst + 3.0f*t10 - 3.0f*t11);
+}
+
+float Camera::get_shutter_curve_nonlinear_normalization_factor() const
+{
+    return 1.0f / m_shutter_curve_nonlinear_close_cdf.evaluate(1.0f);
+}
+
+float Camera::map_to_shutter_curve(const float sample) const
+{
+    assert(sample < 1.0f);
+    if (m_use_nonlinear_shutter_curve)
```

Comment:
You're free to omit the curly braces for single line `if`s. Here using the ternary operator leads to an even more compact notation:

    return m_use_nonlinear_shutter_curve
        ? map_to_shutter_curve_impl_nonlinear(sample)
        : map_to_shutter_curve_impl_linear(sample);

Reply:
Cool;
In cqg we don't omit braces even for one-statement-long body; Habits :)
Fixed.

Reply:
Sure, it's completely correct to keep braces, at my work we also decided to force using them. In appleseed you're free to decide. I just wanted to point out a more compact notation, that's all.